python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -2297,7 +2297,7 @@ def test_deploy_with_consistent_constructor_failure(mock_deployment_state_manage
     _constructor_failure_loop_two_replica(dsm, ds, 3)
 
     assert ds._replica_constructor_retry_counter == 6
-    assert ds.curr_status_info.status == DeploymentStatus.UNHEALTHY
```

Comment:
Do we have other test to cover the case when it's just unhealthy and not deploy failed?

Reply:
yes, there are other tests that cover going from healthy -> unhealthy